FileHandler.gd

```diff
@@ -16,3 +16,61 @@ static func create_directory_if_missing(path: String):
 		var success = dir.make_dir(path)
 		if success != OK:
 			print("Error: Could not create data folder:", path)
+
+static func generate_unique_file_name(base_name: String, extension: String, destination_folder: String) -> String:
+	var unique_name: String = base_name
+	var counter: int = 1
+	var new_file_path: String = destination_folder.plus_file(unique_name + "." + extension)
+
+	while File.new().file_exists(new_file_path):
+		unique_name = base_name + "_" + str(counter)
+		new_file_path = destination_folder.plus_file(unique_name + "." + extension)
+		counter += 1
+	return unique_name + "." + extension
```

Comment:
This function is fine, but it deserves a comment with some input/output examples. This is a prime candidate for some unit tests once we get around to adding those to gdscript code because it is a simple function with strict inputs and outputs.

Reply:
Will add a comment

Reply:
After consideration, I believe it is better design to have the user save the new file. We can default to the old name for the file but this will let the user organize the file if they wish.